daemon/k8s_watcher.go

```diff
@@ -685,6 +685,10 @@ func (d *Daemon) updateK8sNetworkPolicyV1(oldk8sNP, newk8sNP *networkingv1.Netwo
 func (d *Daemon) deleteK8sNetworkPolicyV1(k8sNP *networkingv1.NetworkPolicy) {
 	labels := k8s.GetPolicyLabelsv1(k8sNP)
 
+	if labels == nil {
+		log.Fatalf("provided v1 NetworkPolicy is nil, so cannot delete it")
```

Comment:
let's not be fatal

Reply:
@tgraf suggested to panic earlier.

Reply:
The labels can only be nil because of the new changes. the passed `k8sNP` is never nil.
```
                            if k8sNP := copyObjToV1NetworkPolicy(obj); k8sNP != nil {
						serKNPs.Enqueue(func() error {
							d.deleteK8sNetworkPolicyV1(k8sNP)
							return nil
						}, serializer.NoRetry)
					}
```

Reply:
Why are we not returning errors here?
``` go
                                DeleteFunc: func(obj interface{}) {
                                        metrics.SetTSValue(metrics.EventTSK8s, time.Now())
                                        if k8sNP := copyObjToV1beta1NetworkPolicy(obj); k8sNP != nil {
                                                serKNPs.Enqueue(func() error {
                                                        d.deleteK8sNetworkPolicyV1beta1(k8sNP)
                                                        return nil
                                                }, serializer.NoRetry)
                                        }
                                },
```

Reply:
```
// If `f` returns an error, `waitFunc` will be executed and, depending on the
// return value of `waitFunc`, `f` will be executed again or not.
// The return value of `f` will not be logged and it's up to the caller to log
// it properly.
func (fq *functionQueue) Enqueue(f func() error, waitFunc WaitFunc) {
```

Reply:
OK. This would actually give us an opportunity to log from a central point and have all functions return an error. That makes it very clear when a policy was imported and when it was not imported. I'm fine to do this in another PR though.

Reply:
Created https://github.com/cilium/cilium/issues/3100 to track this.